### ISSUE TITLE
Why matc emits binary-different materials from same source?

### ISSUE THREAD

**Issue by alexey-pelykh** [google/filament#4671]
The sources suggest that due to the multiple jobs running in parallel, yet there's sorting performed after it seems.

**Comment by romainguy** [comment]
I  just tried on several materials and matc produces the same result byte-for-byte every time. You could try without optimizations though, maybe the optimizers sometimes produces slightly different variable names or floating point numbers.

**Comment by alexey-pelykh** [comment]
(1.9.15) `matc -p mobile -a opengl -o test.matc test.mat ; sha256sum test.matc` produces
```
3863e48c1adbde465d885e532855ac4400a55bd1f776e5087220ba7dba585084  test.matc
a8a4b70702ebc56d33e0524a92a86ad4579d14134228d353ccfbe139f5b8cccb  test.matc
...
```
during multiple runs, but 1.12.1 matc seems to produce consistent results. I'd guess something has been fixed after 1.9.15?
